core/src/test/resources/sql/sub-query.iq

```diff
@@ -3448,13 +3448,13 @@ select * from "scott".emp where empno not in (null, 7782);
 
 !ok
 
-EnumerableCalc(expr#0..12=[{inputs}], expr#13=[0:BIGINT], expr#14=[=($t8, $t13)], expr#15=[IS NULL($t12)], expr#16=[>=($t9, $t8)], expr#17=[AND($t15, $t16)], expr#18=[OR($t14, $t17)], proj#0..7=[{exprs}], $condition=[$t18])
+EnumerableCalc(expr#0..12=[{inputs}], expr#13=[0], expr#14=[=($t8, $t13)], expr#15=[IS NULL($t12)], expr#16=[>=($t9, $t8)], expr#17=[AND($t15, $t16)], expr#18=[OR($t14, $t17)], proj#0..7=[{exprs}], $condition=[$t18])
```

Comment:
the disappearance of the cast of 0 may be a hint that the typing is wrong someplace - either the original plan or the new plan must be wrong.

Reply:
I think this indicates that there is a bug somewhere, but it's probably not directly related to these changes, it's just unmasked by these changes.

Reply:
> I think this indicates that there is a bug somewhere, but it's probably not directly related to these changes, it's just unmasked by these changes.

Thanks for the review. I will check this again to make sure we get the right plan.

Reply:
Hi @mihaibudiu, I check the code. These two places build the literal 0 with different methods,
In SqlToRel:
```
final RelDataType longType =typeFactory.createSqlType(SqlTypeName.BIGINT);
final RexLiteral zero =rexBuilder.makeExactLiteral(BigDecimal.ZERO, longType);
```
In SubqueryRemoveRule:
```
builder.literal(0)
```
Because need to compare with the value of COUNT, the first one is more reasonable.

Reply:
yes, the second one seems wrong, since it does not set a type for the zero.

Reply:
I will try to find an SQL test case to test if this is a bug.